wallet/src/service/Offers.ts

```diff
@@ -257,15 +284,41 @@ export const getOfferService = (
   };
 
   const watchPendingOffers = async (brandToPurse: Map<Brand, PurseInfo>) => {
-    const makeProposalEntriesDisplayable = (proposalEntries: {
+    const makeProposalEntriesDisplayable = async (proposalEntries: {
       [key: string]: Amount;
     }) =>
       Object.fromEntries(
-        Object.entries(proposalEntries).map(([kw, entry]) => {
-          const pursePetname = brandToPurse.get(entry.brand)?.pursePetname;
-          const value = String(entry.value);
-          return [kw, { pursePetname, value }];
-        }),
+        await Promise.all(
+          Object.entries(proposalEntries).map(
+            async ([kw, unserializedAmount]) => {
```

Comment:
this callback shares a lot of logic with the above. will they change together? if so please DRY